lib/topology/platforms/shell.py

```diff
@@ -196,6 +201,10 @@ def __init__(
         self._timeout = timeout or -1
         self._encoding = encoding
         self._try_filter_echo = try_filter_echo
+        # Doing this to avoid having a mutable object as default value in the
+        # arguments.
+        if spawn_args is None:
+            self._spawn_args = {'env': {'TERM': 'dumb'}, 'echo': False}
 
```

Comment:
I'll come back to this task soon and I'll let you know when the pull request is ready.